### ISSUE TITLE
var-spacing: False-positive on logical OR (||)

### ISSUE THREAD

**Issue by imrejonk** [ansible/ansible-lint#2241]
##### Summary

The var-spacing check incorrectly reports a logical OR (||) as not having enough spaces.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible --version
ansible [core 2.13.2]
  config file = None
  configured module search path = ['/home/imre/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/imre/.local/lib/python3.9/site-packages/ansible
  ansible collection location = /home/imre/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/imre/.local/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True

$ ansible-lint --version
ansible-lint 6.3.0 using ansible 2.13.2
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Debian 11 GNU/Linux, Python 3.9

##### STEPS TO REPRODUCE

Run ansible-lint against this minimal example playbook:

```
---

- name: Test var-spacing
  hosts: ['localhost']
  gather_facts: False

  tasks:

    - name: Debug
      debug:
        msg: '{{ lookup("pipe", "echo Testing || true") }}'
```

##### Desired Behavior

ansible-lint should not consider a logical OR (||) as a var-spacing issue.

##### Actual Behavior

```
$ ansible-lint test.yml 
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: test.yml
WARNING  Listing 1 violation(s) that are fatal
var-spacing: Jinja2 variables and filters should have spaces before and after.
test.yml:9 Task/Handler: Debug

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - var-spacing  # Jinja2 variables and filters should have spaces before and after.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

##### Additional information

Also see these false-positive issues in the var-spacing check: #2208, #2209

**Comment by ssbarnea** [comment]
PR would be more than welcomed, this might also be a duplicate report.

**Comment by nishipy** [comment]
I'd like to work on this issue.

**Comment by ssbarnea** [comment]
@nishipy It would be better to pick another task because I will likely sort this during the next week. This one is more complex as we will need to switch to jinja2 parser instead of using a regex.
